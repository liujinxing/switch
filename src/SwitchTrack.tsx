import styled from 'styled-components';

const SwitchTrack = styled.span<{ disabled?: boolean; checked?: boolean }>`
  width: 100%;
  height: 100%;
  opacity: ${(props) => (props.theme.palette.type === 'light' ? 0.38 : 0.3)};
  z-index: -1;
  transition: ${(props) =>
    props.theme.transitions.create(['opacity', 'background-color'], {
      duration: props.theme.transitions.duration.shortest,
    })};
  border-radius: 7px;
  background-color: ${(props) =>
    props.theme.palette.type === 'light'
      ? props.theme.palette.black
      : props.theme.palette.white};
`;

export default SwitchTrack;
