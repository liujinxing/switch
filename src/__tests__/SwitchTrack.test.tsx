import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import SwitchTrack from 'src/SwitchTrack';
import TestWrapper from './TestWrapper';

describe('SwitchTrack 镜像测试', () => {
  it('渲染 SwitchTrack', () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <SwitchTrack />
        </TestWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('不可用的SwitchTrack', () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <SwitchTrack disabled />
        </TestWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('选中状态的SwitchTrack', () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <SwitchTrack checked color="info" />
        </TestWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
