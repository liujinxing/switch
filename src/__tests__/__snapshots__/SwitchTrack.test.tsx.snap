// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchTrack 镜像测试 不可用的SwitchTrack 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  opacity: 0.38;
  z-index: -1;
  -webkit-transition: opacity 150ms cubic-bezier(0.4,0.0,0.2,1) 0ms,background-color 150ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: opacity 150ms cubic-bezier(0.4,0.0,0.2,1) 0ms,background-color 150ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  border-radius: 7px;
  background-color: #000000;
  opacity: 0.12;
}

<span
  className="c0"
  disabled={true}
/>
`;

exports[`SwitchTrack 镜像测试 渲染 SwitchTrack 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  opacity: 0.38;
  z-index: -1;
  -webkit-transition: opacity 150ms cubic-bezier(0.4,0.0,0.2,1) 0ms,background-color 150ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: opacity 150ms cubic-bezier(0.4,0.0,0.2,1) 0ms,background-color 150ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  border-radius: 7px;
  background-color: #000000;
}

<span
  className="c0"
/>
`;

exports[`SwitchTrack 镜像测试 选中状态的SwitchTrack 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  opacity: 0.38;
  z-index: -1;
  -webkit-transition: opacity 150ms cubic-bezier(0.4,0.0,0.2,1) 0ms,background-color 150ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: opacity 150ms cubic-bezier(0.4,0.0,0.2,1) 0ms,background-color 150ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  border-radius: 7px;
  background-color: #000000;
  background-color: #00bcd4;
  opacity: 0.5;
}

<span
  checked={true}
  className="c0"
  color="info"
/>
`;
